Doc/library/functools.rst

```diff
@@ -664,6 +669,31 @@ The :mod:`functools` module defines the following functions:
    would have been ``'wrapper'``, and the docstring of the original :func:`example`
    would have been lost.
 
+.. function:: get_variants(func)
+
+   Return all registered function variants for this function. Function variants are
+   objects that represent some subset of the functionality of a function, for
+   example overloads decorated with :func:`typing.overload` or :func:`singledispatch`
+   implementation functions.
+
+   Variants are registered by calling :func:`register_variant`.
+
+   .. versionadded:: 3.11
+
+.. function:: register_variant(func, variant)
+
+   Register *variant* for function *func* that can later be retrieved using
```

Comment:
What are the types for `func` and `variant`? Maybe we need an example of how these three new functions work together?